### DISCUSSION TITLE
Sepformer fine-tuning with custom dataset

### DISCUSSION THREAD

**Discussion by acidcoma** [speechbrain/speechbrain#2025] [Q&A]
Hello everyone! 

I have my own audio separation dataset based on LibriSpeech. I want to train the Sepformer model on my dataset from latest checkpoint, I mean fine-tune existing Sepformer model (with existing weights) from speechbrain to (try to) improve the separation quality. 

Is there any tutorials how to do this? 

If I understand correctly, I have to:
1)  prepare the correct structure of the dataset 
2) correct prepare_data.py file (https://github.com/speechbrain/speechbrain/blob/develop/recipes/LibriMix/prepare_data.py)
3) correct hparams file (https://github.com/speechbrain/speechbrain/blob/develop/recipes/LibriMix/separation/hparams/sepformer-libri2mix.yaml)
4) download model's files (encoder.ckpt, decoder.ckpt, musket.ckpt), also files (optimizer.ckpt, lr_sheduler.ckpt, counter.ckpt)?
5) run train.py script (https://github.com/speechbrain/speechbrain/blob/develop/recipes/LibriMix/separation/train.py)

I'm new to machine learning, help me figure it out please 🙏 

Thanks in advance for your replies and help. :)

**Comment by Adel-Moumen** [comment]
Wdym by "correct hparams file" ? 

Make sure to add the downloaded files in the new save folder that will be created when you will launch the experiment. 

Everything else looks good.
